modules/siddhi-core/src/main/java/io/siddhi/core/table/record/AbstractQueryableRecordTable.java

```diff
@@ -91,14 +99,25 @@
 public abstract class AbstractQueryableRecordTable extends AbstractRecordTable implements QueryableProcessor {
     private static final Logger log = Logger.getLogger(AbstractQueryableRecordTable.class);
     private int maxCacheSize;
-    private InMemoryTable cachedTable;
-    private boolean isCacheEnabled = false;
+    private long purgeInterval;
+    private boolean cacheEnabled = false;
+    private boolean cacheExpiryEnabled = false;
+    private InMemoryTable cacheTable;
     private CompiledCondition compiledConditionForCaching;
     private CompiledSelection compiledSelectionForCaching;
     private Attribute[] outputAttributesForCaching;
-    private TableDefinition cacheTableDefinition;
-    private SiddhiAppContext siddhiAppContext;
+    protected SiddhiAppContext siddhiAppContext;
+    protected StateEvent findMatchingEvent;
+    protected Selector selectorForTestStoreQuery;
+    protected SiddhiQueryContext siddhiQueryContextForTestStoreQuery;
+    protected MatchingMetaInfoHolder matchingMetaInfoHolderForTestStoreQuery;
+    protected StateEvent containsMatchingEvent;
     private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+    private long storeSizeLastCheckedTime;
+    private long storeSizeCheckInterval;
+    private long retentionPeriod;
+    private long cacheLastReloadTime;
+    public static ThreadLocal<Boolean> queryFromStore = ThreadLocal.withInitial(() -> Boolean.FALSE);
```

Comment:
queryFromStore does not give the meaning of why this ThreadLocal is used, improve its name.